libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/KotlinPlugin.kt

```diff
@@ -998,19 +999,18 @@ abstract class AbstractAndroidProjectHandler(private val kotlinConfigurationTool
         processAndroidKotlinAndJavaSources(
             compilation,
             addKotlinSources = { kotlinSourceSet -> compilation.source(kotlinSourceSet) },
-            addJavaSources = { sources -> compilation.compileKotlinTaskProvider.configure { it.source(sources) } }
+            addJavaSources = { sources ->
+                compilation.compileKotlinTaskProvider.configure { it.source(sources.dir) }
```

Comment:
This is not used to track task dependency info, as params get unwrapped to get the sources.

Reply:
I see `it.source(..)` calls `KotlinCompile#source`, which calls `org.gradle.api.tasks.SourceTask#source` where the given source should usually contain both the file location and the task dependency info, otherwise the source file may not be available/up-to-date when `KotlinCompile` runs.

It's not an issue introduced by this PR, but I wanted to check if the task dependency has been handled properly yet. (Feel free to ignore if you think it's out of the scope of this PR.)

Reply:
There are still some `dependsOn` invocations in the KGP code base that I think we should clean up separately. Let's do that in another PR.

Reply:
SGTM!